packages/backend/src/graphql/schema.graphql

```diff
@@ -770,6 +784,31 @@ type PaymentPointerWithdrawal {
   paymentPointer: PaymentPointer!
 }
 
+type WebhookEvent implements Model {
+  "Event id"
+  id: ID!
+  "Type of event"
+  type: String!
+  "Stringified JSON data"
+  data: String!
+  "Date-time of creation"
+  createdAt: String!
+}
+
+type WebhookEventsConnection {
+  pageInfo: PageInfo!
+  edges: [WebhookEventsEdge!]!
+}
+
+type WebhookEventsEdge {
+  node: WebhookEvent!
+  cursor: String!
+}
+
+input WebhookEventFilter {
+  type: String
+}
```

Comment:
I am trying to determine the pattern to set with this initial filter. I've been looking at this version but I'm not sure if we want or need the full extent of this filtering: https://medium.com/@conrardy/expanding-relay-cursor-connections-cb6b84fee1e8

It seems pretty comprehensive but I'm not sure to what extent we would want to implement this in this task. Im trying to think about simplified versions (which could be extended further if needed).

Currently the `WebhookEventFilter` assumes an "and" comparison with each filter. For `type`, it assumes an equality comparison. Additional filters wouldn't need to be strictly equal.

I'm not sure if this is extendable enough.  If we wanted to also get events with withdrawalAmount > 0, for example, we could do:

**A)** (current way)
```gql
input WebhookEventFilter {
  type: String
  withdrawalAmount_gt: UInt64
}
```

or maybe

**B)**
```gql
input WebhookEventFilter {
  type: {eq: String, contains: String, startsWith: String etc.}
  withdrawalAmount: {gt: UInt64, lt: UInt64, eq: Uint64 etc.}
}
```

Although in either **A** or **B**, nothing is stopping someone from making a request with all these comparisons. In that case it seems they are assumed to all apply (treating them as "and"). So someone could request all withdrawalAmounts > 0 and equal to -1. Shame on them I guess, but this makes me wonder if there is a better pattern. This is basically 1 step short of the referenced medium article, which takes the additional leap of implementing and/or with a filter like this:

**C)**
```gql
input WebhookEventFilter {
    and: [WebhookEventFilter!]
    or: [WebhookEventFilter!]
    type: {eq: String, contains: String, startsWith: String etc.}
    withdrawalAmount: {gt: UInt64, lt: UInt64, eq: Uint64 etc.}
}
```

Currently I'm leaning towards **B** (without the example extensions), which would look something like:

```gql
input WebhookEventFilter {
  type: {eq: String}
}
```

 This would involve creating some utility function to apply the filter to a knex query builder. If we ever wanted to move to **C** we would add the `and` and `or` fields to schema (trivial, non-breaking) and then implement the logic in the utility function (more complicated). I think there is also an argument to be made for **A** because it's very simple and everything we need right now, but if we intend on extending it more I prefer to start with that pattern.

Reply:
I like the proposed solution for B. Keeps this pretty simple at the beginning, and leaves the door (relatively) open for any additional filters. I might even suggest something like:

```gql
input WebhookEventFilter { 
    type: {in: String[]} 
}
```
Seems pretty useful in the context of the admin API, which could basically have a box for selecting multiple types at once.

Reply:
`in` makes sense instead of `eq`. Works for 1 or many. Good suggestion.